### PR TITLE
CAL-80 add precision reducer to JTS geometry

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Add precision reducer to JTS geometry objects.
#### Who is reviewing it?
@jrnorth 
@rzwiefel 
@lcrosenbu 
@bdeining 
#### How should this be tested?
Ingest a video with known precision problems (contact me for the file). Verify that the file is ingested.
#### What are the relevant tickets?
CAL-80
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -36,6 +36,8 @@
         </argument>
     </bean>
 
+    <bean id="reduce" class="org.codice.alliance.libs.klv.GeometryReducer"/>
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: This ID is a little vague, maybe `geometryReducer`?

**Comment by bdeining** [comment]
LGTM

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 1 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: ok, i'll change it

File: libs/klv/src/main/java/org/codice/alliance/libs/klv/LocationKlvProcessor.java [hunk 2]
```diff
@@ -39,22 +39,28 @@
...
 
     /**
-     * @param geometryFunction transform the final Geometry object (must be non-null)
+     * @param postUnionGeometryOperator transform the final Geometry object (must be non-null)
+     * @param preUnionGeometryOperator  transform the final Geometry object (must be non-null)
      */
-    public LocationKlvProcessor(GeometryFunction geometryFunction) {
-        notNull(geometryFunction, "geometryFunction must be non-null");
-        this.geometryFunction = geometryFunction;
+    public LocationKlvProcessor(GeometryOperator postUnionGeometryOperator,
```
Comment (jrnorth) [inline libs/klv/src/main/java/org/codice/alliance/libs/klv/LocationKlvProcessor.java]: Seems like the parameter order should be reversed.

**Comment by bdeining** [comment]
Built and tested with a video with known precision problems.  Observed some rendering problems related to MULTIPOLYGON geometries unrelated to the PR.  https://codice.atlassian.net/browse/DDF-1804.  LGTM

**Comment by jaymcnallie** [comment]
👍

**Pull request merged**